system-tests/src/main/java/io/apicurio/registry/systemtests/framework/KeycloakUtils.java

```diff
@@ -66,25 +80,24 @@ public static void deployKeycloak(ExtensionContext testContext, String namespace
                 "-f", getKeycloakFilePath("keycloak-realm.yaml")
         );
 
-        // TODO: Wait for Keycloak Realm readiness, but API model not available
-
         LOGGER.info("Keycloak should be deployed.");
     }
 
-    public static void removeKeycloak() {
-        removeKeycloak(Constants.TESTSUITE_NAMESPACE);
-    }
-
-    public static void removeKeycloak(String namespace) {
-        LOGGER.info("Removing Keycloak...");
+    public static void removeKeycloakRealm(String namespace) {
+        LOGGER.info("Removing keycloak realm");
 
         Exec.executeAndCheck(
                 "oc",
                 "delete",
                 "-n", namespace,
                 "-f", getKeycloakFilePath("keycloak-realm.yaml")
         );
+    }
 
+    public static void removeKeycloak(String namespace) throws InterruptedException {
+        removeKeycloakRealm(namespace);
+        Thread.sleep(Duration.ofMinutes(2).toMillis());
```

Comment:
Hey, it should but many times it happens that the realm is not deleted before keycloak and than the namespace is stuck in the terminating phase because of the realm finalizer. Yes, in an ideal world we should add resources for Keycloak objects and create some more intelligent wait ...